LdapPatch/upgradeLdap.pl

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/perl
+
+use strict;
+
+use Cwd;
+use Time::localtime qw(ctime);
+
+our $platform = qx(/opt/zimbra/libexec/get_plat_tag.sh);
+chomp $platform;
+
+my $logFileName = "zmsetup.".getDateStamp().".log";
+my $logfile = "/tmp/".$logFileName;
+open LOGFILE, ">$logfile" or die "Can't open $logfile: $!\n";
+unlink("/tmp/zmsetup.log") if (-e "/tmp/zmsetup.log");
+symlink($logfile, "/tmp/zmsetup.log");
+
+my $ol = select (LOGFILE);
+select ($ol);
+$| = 1;
+
+progress("Operations logged to $logfile\n");
+
+our $ZMPROV = "/opt/zimbra/bin/zmprov -r -m -l";
+our $SU = "su - zimbra -c ";
+
+my $pwuid = getpwuid( $< );
+if ($pwuid eq "zimbra") {
+  $SU = "";
+}
+
+our %options = ();
+
+if (isInstalled("zimbra-ldap")) {
+  installLdapSchema();
+}
+
+sub installLdapSchema {
+  main::runAsZimbra("/opt/zimbra/libexec/zmldapschema 2>/dev/null");
+}
+
+my $rc = runAsZimbra ("/opt/zimbra/libexec/zmldapupdateldif");
+
+sub isInstalled {
+  my $pkg = shift;
+
+  my $pkgQuery;
+
+  my $good = 0;
+  if ($platform =~ /^DEBIAN/ || $platform =~ /^UBUNTU/) {
+    $pkgQuery = "dpkg -s $pkg";
+  } else {
+    $pkgQuery = "rpm -q $pkg";
+  }
+
+  my $rc = 0xffff & system ("$pkgQuery > /dev/null 2>&1");
+  $rc >>= 8;
+  if (($platform =~ /^DEBIAN/ || $platform =~ /^UBUNTU/) && $rc == 0 ) {
+    $good = 1;
+    $pkgQuery = "dpkg -s $pkg | egrep '^Status: ' | grep 'not-installed'";
+    $rc = 0xffff & system ("$pkgQuery > /dev/null 2>&1");
+    $rc >>= 8;
+    return ($rc == $good);
+  } else {
```

Comment:
else is not required, can just return ($src == good) after if block

Reply:
done